addon/templates/components/text-field.hbs

```diff
@@ -22,13 +22,23 @@
     type=type
     class=(concat "cs-component-text-field--input " @theme)
     required=required
-    _value=value
+    value=value
     onblur=onblur
     onchange=onchange
     oninput=oninput
     id=inputId
     data-test-cs-component-input=dataTestName
   }}
+{{else if otherInputComponent}}
+  {{component otherInputComponent
+    class=(concat "cs-component-text-field--input " @theme)
+    required=required
+    value=value
+    oninput=(action "validate")
+    onblur=(action "validate")
+    id=inputId
+    data-test-cs-component-input=dataTestName
+  }}
```

Comment:
It seems weird to introduce an `otherInputComponent` here. Were there issues using `inputComponent`? We may have to refactor so that we can pass custom attributes to `inputComponent` and set them from whatever child class of `TextField`.

Right now the attributes being passed in to `inputComponent` are specific to the `phone-number-field/input` component. What do you think of refactoring so that `email` does something like this:

```javascript
// email.js

export default TextField.extend({
  inputComponent: 'email-field/input',
  inputComponentOptions: {
    oninput: this.actions.validate,
    // ...
  }
});
```

At least with this approach, we can keep the parameters passed into `inputComponent` standard (in the hbs) and then have the custom ones in `inputComponentOptions`. This will require some refactoring but just tossing this out as a potential approach so that we don't have `otherInputComponent`, `otherInputComponent2`, etc. Thoughts?

Reply:
Yes, that's the feedback I was looking for. Maybe we can pair on that tomorrow or next week and see how we can combine those various types of input fields together. Also I believe you were saying that you did some refactoring for the main text input element to use ember input component, so it would be good to have those changes in as well.